Measurements/Measurements.ipynb

```diff
@@ -0,0 +1,571 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%kata T105_ZeroZeroOrOneOne_Test\n",
+    "\n",
+    "operation ZeroZeroOrOneOne (qs : Qubit[]) : Int {\n",
+    "    // ...\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Task 1.6. Distinguish four basis states.\n",
+    "\n",
+    "**Input:** Two qubits (stored in an array of length 2) which are guaranteed to be in one of the four basis states ($|00\\rangle$, $|01\\rangle$, $|10\\rangle$, or $|11\\rangle$).\n",
+    "\n",
+    "**Output:**\n",
+    "\n",
+    "* 0 if the qubits were in the $|00\\rangle$ state,\n",
+    "* 1 if they were in the $|01\\rangle$ state, \n",
+    "* 2 if they were in the $|10\\rangle$ state, \n",
+    "* 3 if they were in the $|11\\rangle$ state.\n",
+    "\n",
+    "In this task and the subsequent ones the order of qubit states in task description matches the order of qubits in the array (i.e., $|10\\rangle$ state corresponds to `qs[0]` in state $|1\\rangle$ and `qs[1]` in state $|0\\rangle$).\n",
+    "\n",
+    "The state of the qubits at the end of the operation does not matter."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%kata T106_BasisStateMeasurement_Test\n",
+    "\n",
+    "operation BasisStateMeasurement (qs : Qubit[]) : Int {\n",
+    "    // ...\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Task 1.7. Distinguish two basis states given by bit strings\n",
+    "\n",
+    "**Inputs:** \n",
+    "\n",
+    "1. $N$ qubits (stored in an array of length $N$) which are guaranteed to be in one of the two basis states described by the given bit strings.\n",
+    "2. Two bit strings represented as `Bool[]`s.\n",
+    "\n",
+    "**Output:** \n",
+    "\n",
+    "* 0 if the qubits were in the basis state described by the first bit string,\n",
+    "* 1 if they were in the basis state described by the second bit string.\n",
+    "\n",
+    "Bit values `false` and `true` correspond to $|0\\rangle$ and $|1\\rangle$ states. You are guaranteed that both bit strings have the same length as the qubit array, and that the bit strings differ in at least one bit.\n",
+    "\n",
+    "You can use exactly one measurement. The state of the qubits at the end of the operation does not matter.\n",
+    "\n",
+    "> Example:  for bit strings `[false, true, false]` and `[false, false, true]` return 0 corresponds to state $|010\\rangle$, and return 1 corresponds to state $|001\\rangle$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%kata T107_TwoBitstringsMeasurement_Test\n",
+    "\n",
+    "operation TwoBitstringsMeasurement (qs : Qubit[], bits1 : Bool[], bits2 : Bool[]) : Int {\n",
```

Comment:
The Q# style guide suggests that qubit arguments should come last, and that operation names should be verbs (e.g.: `MeasureTwoBitstrings`).